dev/AppNotifications/AppNotificationTelemetry.h

```diff
@@ -0,0 +1,319 @@
+        }
+    }
+    CATCH_LOG()
+
+    DEFINE_EVENT_METHOD(RemoveByTagAndGroupAsyncByAPI)(
+        winrt::hresult hr,
+        std::wstring const& appId) noexcept try
+    {
+        if (c_maxEventLimit >= UpdateLogEventCount())
+        {
+            TraceLoggingClassWriteMeasure(
+                "RemoveByTagAndGroupAsyncByAPI",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
+                TraceLoggingHexUInt32(hr, "OperationResult"),
+                TraceLoggingWideString(appId.c_str(), "AppId"),
+                TraceLoggingBool(IsPackagedApp(), "IsAppPackaged"),
+                TraceLoggingWideString(GetAppName(), "AppName"));
+        }
+    }
+    CATCH_LOG()
+
+    DEFINE_EVENT_METHOD(RemoveByGroupAsyncByAPI)(
+        winrt::hresult hr,
+        std::wstring const& appId) noexcept try
+    {
+        if (c_maxEventLimit >= UpdateLogEventCount())
+        {
+            TraceLoggingClassWriteMeasure(
+                "RemoveByGroupAsyncByAPI",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
+                TraceLoggingHexUInt32(hr, "OperationResult"),
+                TraceLoggingWideString(appId.c_str(), "AppId"),
+                TraceLoggingBool(IsPackagedApp(), "IsAppPackaged"),
+                TraceLoggingWideString(GetAppName(), "AppName"));
+        }
+    }
+    CATCH_LOG()
+
+    DEFINE_EVENT_METHOD(RemoveAllAsyncByAPI)(
+        winrt::hresult hr,
+        std::wstring const& appId) noexcept try
+    {
+        if (c_maxEventLimit >= UpdateLogEventCount())
+        {
+            TraceLoggingClassWriteMeasure(
+                "RemoveAllAsyncByAPI",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
+                TraceLoggingHexUInt32(hr, "OperationResult"),
+                TraceLoggingWideString(appId.c_str(), "AppId"),
+                TraceLoggingBool(IsPackagedApp(), "IsAppPackaged"),
+                TraceLoggingWideString(GetAppName(), "AppName"));
+        }
+    }
+    CATCH_LOG()
+
+    DEFINE_EVENT_METHOD(GetAllAsyncByAPI)(
+        winrt::hresult hr,
+        std::wstring const& appId) noexcept try
+    {
+        if (c_maxEventLimit >= UpdateLogEventCount())
+        {
+            TraceLoggingClassWriteMeasure(
+                "GetAllAsyncByAPI",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
+                TraceLoggingHexUInt32(hr, "OperationResult"),
+                TraceLoggingWideString(appId.c_str(), "AppId"),
+                TraceLoggingBool(IsPackagedApp(), "IsAppPackaged"),
+                TraceLoggingWideString(GetAppName(), "AppName"));
+        }
+    }
+    CATCH_LOG()
+
+private:
+    Microsoft::WRL::Wrappers::CriticalSection m_lock;
```

Comment:
Nit: It is encouraged to use wil wrappers or winrt APIs instead of WRL.

Reply:
wil::srwlock m_lock in most of our classes :)

Reply:
Yep. Shouldn't be using WRL. 
wil == good!

Reply:
Done, fixed in PushNotificationsTelemetry also